hack/loadtest/util/placement.go

```diff
@@ -251,39 +264,49 @@ func waitForCrpToComplete(ctx context.Context, hubClient client.Client, deadline
+			klog.ErrorS(err, "failed to get crp", "crp", crpName)
+		}
+		appliedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementAppliedConditionType))
+		synchronizedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementSynchronizedConditionType))
+		scheduledCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementScheduledConditionType))
 		select {
 		case <-ctx.Done():
 			// Context has been cancelled, exit the loop
-			return
-		case <-timer.C:
-			// Deadline has been reached, exit the loop
-			klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
+			klog.V(3).Infof("the cluster resource placement `%s` timeout", crpName)
 			LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
 			updateTimeoutCount.Inc()
 			return
-		case <-ticker.C:
-			// Poll for the CRP status
-			if err = hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: ""}, &crp); err != nil {
-				klog.ErrorS(err, "failed to get crp", "crp", crpName)
-				continue
+		case <-timer.C:
+			// Deadline has been reached
+			if condition.IsConditionStatusTrue(appliedCond, 1) && condition.IsConditionStatusTrue(synchronizedCond, 1) && condition.IsConditionStatusTrue(scheduledCond, 1) {
+				// succeeded
+				klog.V(3).Infof("the cluster resource placement `%s` succeeded", crpName)
+				LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "succeed").Inc()
+				updateSuccessCount.Inc()
+				LoadTestUpdateLatencyMetric.WithLabelValues(currency, fleetSize).Observe(time.Since(deletionStartTime).Seconds())
+				return
 			}
-			// check if the condition is true
-			appliedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementAppliedConditionType))
-			synchronizedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementSynchronizedConditionType))
-			scheduledCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementScheduledConditionType))
+			// failed
+			klog.V(3).Infof("the cluster resource placement `%s` failed", crpName)
+			LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "failed").Inc()
+			updateFailCount.Inc()
+			return
+
+		case <-ticker.C:
+			// Interval for CRP status check
 			if condition.IsConditionStatusTrue(appliedCond, 1) && condition.IsConditionStatusTrue(synchronizedCond, 1) && condition.IsConditionStatusTrue(scheduledCond, 1) {
 				// succeeded
 				klog.V(3).Infof("the cluster resource placement `%s` succeeded", crpName)
 				LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "succeed").Inc()
 				updateSuccessCount.Inc()
 				LoadTestUpdateLatencyMetric.WithLabelValues(currency, fleetSize).Observe(time.Since(deletionStartTime).Seconds())
 				return
+			} else if appliedCond == nil || appliedCond.Status == metav1.ConditionUnknown || synchronizedCond == nil || synchronizedCond.Status == metav1.ConditionUnknown || scheduledCond == nil || scheduledCond.Status == metav1.ConditionUnknown {
```

Comment:
Hi Britania! A nit: it seems to be that for simplicity, an `if ... else if ... else` would do here.